tools/generate.py

```diff
@@ -940,6 +980,27 @@ def start(module: str, version: str, overrides: dict[str, str]) -> str:
 
     args = parser.parse_args()
 
+    ns = {
+        "core": "http://www.gtk.org/introspection/core/1.0",
+        "c": "http://www.gtk.org/introspection/c/1.0",
+        "glib": "http://www.gtk.org/introspection/glib/1.0",
+    }
+    gir_tree = ET.parse(f"/usr/share/gir-1.0/{args.module}-{args.version}.gir")
```

Comment:
This will not work with `gir`s installed to other locations (e.g. on Nix). We should at least support `XDG_DATA_DIRS`. But probably better to follow the procedure from g-i:

https://gitlab.gnome.org/GNOME/gobject-introspection/-/merge_requests/258
https://gitlab.gnome.org/GNOME/gobject-introspection/-/blob/f31a2b5c30b3a32e1474ed26e560bdb795bbbd2c/giscanner/transformer.py#L183-206

Reply:
I am unable to test this, please check if it works

Reply:
Thanks. Looks great. I verified that it works with #171.